### PR TITLE
WIP: Update the collectives introduction for teams

### LINKED ISSUE #46: Revise collectives introduction
The introduction to the collectives API needs revision:

- [x] Add intro text for team-based collectives
- [x] Add team-based API listing
- [x] Deprecate active set-based API listing (see #40)
- [ ] Specify blocking/nonblocking semantics for collectives (see #44, openshmem-org/specification#228)

### PR DESCRIPTION
**Pull request by nspark**
This PR attempts to update the collectives introduction for the new teams API. Currently, this draft includes the following (but definitely needs review):

- [x] Add intro text for team-based collectives
- [ ] Add team-based API listing
- [x] Deprecate active set-based API listing (see #40)
- [x] Specify blocking/nonblocking semantics for collectives (see #44, openshmem-org/specification#228)

Closes #46

### CODE REVIEW

**Comment by nspark** [comment]
[PDF of PR as of `077a8e3`](https://github.com/gmegan/specification/files/2431147/openshmem-draft-077a8e3.pdf)

File: content/collective_intro.tex [hunk 1]
```diff
@@ -16,29 +60,29 @@
 satisfy the requirement that its last member corresponds to a valid \ac{PE}
 number, that is
 $0 \le PE\_start + (PE\_size - 1) * 2^{logPE\_stride} < npes$.
-All \acp{PE} participating in the collective routine must provide the same
+All \acp{PE} participating in the \newtext{active set-based} collective routine must provide the same
```
Comment (gmegan) [inline content/collective_intro.tex:63]: The team-based, active set-based is a bit hard to parse. Changing from "active set-based" to "active-set-based" may be more readable for this phrasing. Or even "non-teams collectives" could work. Also, some structural changes may help, e.g. putting discussion of active sets in a separate subsection that can be easily deprecated out.

File: content/collective_intro.tex [hunk 2]
```diff
@@ -1,11 +1,55 @@
...
+
+The team-based collective routines are performed with respect to a valid
+\openshmem team, which is specified by a team handle argument.
+Team-based collective operations require all \acp{PE} in the team to call
+the routine in order for the operation to complete.
+
+A team-based collective operation is considered \emph{active} on a \ac{PE}
+when that \ac{PE} calls one of the team-based collective routines.
+Such operations may be blocking or nonblocking and the nature of which
+determines the conditions under which the operation is considered complete:
```
Comment (gmegan) [inline content/collective_intro.tex:16]: Nonblocking collectives discussion would be easier to add as a separate PR from this one, since the notion of nonblocking collectives is still under discussion.

File: content/collective_intro.tex [hunk 3]
```diff
@@ -47,3 +91,17 @@
 \item \FUNC{shmem\_alltoall\{32, 64\}}
 \item \FUNC{shmem\_alltoalls\{32, 64\}}
 \end{itemize}
+\end{DeprecateBlock}
+
+\newtext{
+Additionally, \openshmem provides two synchronization routines for backward
+compatibility, which are defined with semantic equivalence to team- and
+context-based routines:
```
Comment (gmegan) [inline content/collective_intro.tex:99]: If this text is to be added in this PR, should we also go ahead and change the individual descriptions of these routines in the same PR?

File: content/collective_intro.tex (hunk 1 above)
Comment (nspark) [reply content/collective_intro.tex:63]: > The team-based, active set-based is a bit hard to parse.

👍 on "active-set-based". I'm not crazy about "non-teams collectives" because routines like `shmem_barrier_all()` would fall under that, and I'd like to handle those separately (e.g., by defining them in terms of team-based collectives).

The main reason I didn't make an explicit subsection for discussion of active sets was that no other API section introduction has subsections. It wouldn't be a terrible thing, but I was trying stay consistent with the existing document structure. With the existing deprecation markings, I think it's pretty clear about which text is deprecated.

File: content/collective_intro.tex (hunk 3 above)
Comment (nspark) [reply content/collective_intro.tex:99]: Ah, good point. I'll have to make those changes, too.

File: content/collective_intro.tex (hunk 1 above)
Comment (gmegan) [reply content/collective_intro.tex:63]: Agreed about the point on "non-teams" being misleading due to the "*_all" collectives being team-based under the covers. Also agreed that deprecation markings should be good enough. Plus this avoids actually adding new structure and content into something that is going to be removed. I think it would help to have a sentence or two at the top of the intro to mention that there are two types of collectives. I will add an in-line comment there....

File: content/collective_intro.tex [hunk 4]
```diff
@@ -1,11 +1,55 @@
...
-operations on a group of \acp{PE} called an active set. The collective
-routines require all \acp{PE} in the active set to simultaneously call the
-routine.  A \ac{PE} that is not in the active set calling the collective
-routine results in undefined behavior.  All collective routines have an
+\emph{Collective routines} are defined as \newtext{coordinated} communication or
+synchronization operations performed \newtext{by} \oldtext{on} a group of \acp{PE} \oldtext{called an active set}.
+{\color{ForestGreen} % <start-newtext>
+\openshmem provides collective routines that operate on teams
+(see Section~\ref{subsec:team} on team management)
+\DeprecateInline{and active sets}.
```
Comment (gmegan) [inline content/collective_intro.tex:6]: It would help set up for subsequent discussion to have a bit more text here. Something like:

OpenSHMEM provides collective routines of two types
- Collective routines that operate on teams use a team handle argument to determine which PEs will participate in the routine, and use resources internal to the team to perform operations. These routines are to be the standard for OpenSHMEM moving forward.
- Collective routines that operate on active sets use a set of parameters to determine which PEs will participate and what resources are used to perform operations. These routines are the legacy API for collectives which will be deprecated and phased out of implementations moving forward.

**Comment by gmegan** [comment]
Some text was copied over into a new branch and a new PR was opened to complete the issue.
